### PR TITLE
[DEV-1830] target_namespace: create target namespace when we create a target

### PR DESCRIPTION
**Pull request by jevonyeoh**
## Description
Add ability to create a target namespace when we create a target.

Will follow up with the update paths, and creating a new version when a target with the same name exists.

## Related Issue
https://featurebyte.atlassian.net/browse/DEV-1830

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: featurebyte/service/feature.py [hunk 1]
```diff
@@ -72,31 +73,6 @@ async def validate_feature_version_and_namespace_consistency(
             )
 
 
-def sanitize_query_graph_for_feature_definition(graph: QueryGraphModel) -> QueryGraphModel:
```
Comment (jevonyeoh) [inline featurebyte/service/feature.py:75]: moved to separate module so can reuse

File: featurebyte/service/target.py [hunk 2]
```diff
@@ -14,3 +70,129 @@ class TargetService(BaseDocumentService[TargetModel, TargetCreate, TargetService
...
+                    data=TargetNamespaceCreate(
+                        _id=document.target_namespace_id,
+                        name=document.name,
+                        target_ids=[insert_id],
+                        default_target_id=insert_id,
+                        default_version_mode=DefaultVersionMode.AUTO,
+                        entity_ids=sorted(entity_ids),
+                    ),
+                )
+        return await self.get_document(document_id=insert_id)
```
Comment (gee-senbong) [inline featurebyte/service/target.py:198]: Seems like this part shares many common logics with `FeatureService` class. Would be good if we could reuse the logic.

Comment (gee-senbong) [reply featurebyte/service/target.py:198]: If we could reuse the logic, the tests in https://github.com/featurebyte/featurebyte/blob/b8ac5141588339ca221a0b952437cae810e1e176/tests/unit/service/test_feature.py will cover some of the logic here.

Comment (jevonyeoh) [reply featurebyte/service/target.py:198]: hm.. let me try to refactor some of it. am also planning to refactor a bit more all at once when i've added all the logic in (for the update code paths as well), mostly wanted to try to get it to a working state first heh.

**Review by gee-senbong — APPROVED** [review]
LGTM.  We could refactor the duplicated codes between feature service & target service later.

**Comment by jevonyeoh** [comment]
i've stubbed out a base service for now, will refactor abit more in a follow-up!!

**Pull request merged**